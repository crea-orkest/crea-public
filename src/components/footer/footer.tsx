import React from 'react'
import classNames from 'classnames'
import styles from './styles.module.scss'

export const Footer = () => {
  return (
    <footer className={classNames(styles.wrapper)}>
      <ul>
        <li>Concerten</li>
      </ul>
      <ul>
        <li>Instagram</li>
        <li>Facebook</li>
      </ul>
    </footer>
  )
}
