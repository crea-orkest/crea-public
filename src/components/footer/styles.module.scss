@use 'sass:map';
@use '../../styles/tokens/colors.scss' as colors;
@use '../../styles/tokens/layout.scss' as layout;
@use '../../styles/tokens/spacing.scss' as spacing;
@use '../../styles/tokens/breakpoints.scss' as breakpoints;

.wrapper {
  background-color: colors.$primary;
  color: colors.$background;
  padding-top: map.get(spacing.$rem, 48);
  padding-bottom: map.get(spacing.$rem, 48);
  align-self: end;

  @media screen and (min-width: breakpoints.$sm) {
    padding-top: map.get(spacing.$rem, 64);
    padding-bottom: map.get(spacing.$rem, 64);
  }
}

.content {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  gap: map.get(spacing.$rem, 32);

  @media screen and (min-width: breakpoints.$sm) {
    flex-direction: row;
  }
}

.title {
  color: colors.$secondary-light;
  margin-bottom: map.get(spacing.$rem, 8);
}

.address {
  font-style: normal;
  margin-bottom: map.get(spacing.$rem, 32);
}

.link {
  color: colors.$light;
}

.list {
  list-style: none;
  padding: 0;
  margin: 0;
}

.logoList {
  display: flex;
  flex-direction: column;
  gap: map.get(spacing.$rem, 16);
}

.logo {
  position: relative;
  object-fit: contain;
  object-position: left;
}

.logoLink {
  display: block;
  position: relative;
  width: 100%;
  height: 30px;

  @media screen and (min-width: breakpoints.$sm) {
    height: 40px;
  }
}
