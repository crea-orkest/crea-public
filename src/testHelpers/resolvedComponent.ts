export const resolvedComponent = async <T>(
  Component: Function,
  props: T
): Promise<() => JSX.Element> => {
  const ComponentResolved = await Component(props);
  return () => ComponentResolved;
};
