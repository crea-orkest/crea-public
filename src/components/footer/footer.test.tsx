import { Footer } from './footer'
import React from 'react'
import { render } from '@testing-library/react'
import { resolvedComponent } from 'utils/testHelpers/resolvedComponent'

jest.mock('../../graphql/getters/getSiteMetadata', () => {
  const originalModule = jest.requireActual(
    '../../graphql/getters/getSiteMetadata'
  )
  return {
    __esModule: true,
    ...originalModule,
    getSiteMetadata: jest.fn(() => ({
      metadata: {
        title: 'Default title',
        description: 'Default description',
        base_url: 'https://example.com',
      },
      error: undefined,
    })),
  }
})

describe('Footer component', () => {
  it('shows all the data', async () => {
    const ResolvedFutureEvents = await resolvedComponent(Footer, {})

    const { container } = render(<ResolvedFutureEvents />)

    expect(container).toMatchSnapshot()
  })
})
