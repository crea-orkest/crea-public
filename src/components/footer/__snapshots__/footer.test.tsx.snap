// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component shows all the data 1`] = `<div />`;
